### PR TITLE
Ignore small regressions for JavaIDEModelPerformanceTest

### PR DESCRIPTION
**Pull request by wolfs**
The test is very flaky, no matter how much we rebaseline it. Let's ignore any regressions below 5% for it.

This PR also ignores regressions below 10ms median difference and always detects regressions above 20% median difference, no matter the confidence.

### CODE REVIEW

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy [hunk 1]
```diff
@@ -82,7 +105,16 @@ class BaselineVersion implements VersionResults {
...
 
     private static boolean differenceIsSignificant(DataSeries<Duration> myTime, DataSeries<Duration> otherTime, double minConfidence) {
-        DataSeries.confidenceInDifference(myTime, otherTime) > minConfidence
+        return (myTime.median - otherTime.median).abs() > MINIMUM_DIFFERENCE_WE_CAN_MEASURE &&
+            (relativeDifferenceInMedianIsVeryHigh(myTime, otherTime) || DataSeries.confidenceInDifference(myTime, otherTime) > minConfidence)
     }
 
+    private static boolean differenceInMedianIsSignificant(DataSeries<Duration> myTime, DataSeries<Duration> otherTime, double minRelativeMedianDifference) {
+        def minimumMedian = [myTime.median, otherTime.median].min()
+        return ((myTime.median - otherTime.median) / minimumMedian).abs() > minRelativeMedianDifference
```
Comment (asodja) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy:114]: does this handle also zero or some such corner values?

Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy:114]: It does not. Do you think handling of 0 is necessary? That would mean we have a Gradle invocation which takes 0 milliseconds every time.

**Review by asodja — APPROVED** [review]
LGTM

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy (hunk 1 above)
Comment (asodja) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy:114]: I was thinking if maybe measurement code would report 0, due to some bug, or some other weird case

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**